File: apps/apple/Shared/Sources/UI/ServerStatusView.swift
import SwiftUI

// run-state indicator: a tinted halo around a solid dot
struct ServerStatusView: View {
    let isRunning: Bool

    private var dotColor: Color {
        isRunning ? Theme.running : Color(hex: 0xBDBDBD)
    }

    var body: some View {
        VStack(spacing: 12) {
            ZStack {
                Circle()
                    .fill(dotColor.opacity(0.12))
                    .frame(width: 56, height: 56)

                Circle()
                    .fill(dotColor)
                    .frame(width: 20, height: 20)
            }

            Text(isRunning ? "Running" : "Stopped")
                .font(.subheadline.weight(.semibold))
                .foregroundStyle(isRunning ? Theme.success : Theme.muted)
        }
    }
}

File: apps/android/app/src/main/java/com/ionclaw/app/ui/ServerStatusView.kt
package com.ionclaw.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.ionclaw.app.ui.theme.BrandRunning
import com.ionclaw.app.ui.theme.BrandSuccess
import com.ionclaw.app.ui.theme.MutedText

// run-state indicator: a tinted halo around a solid dot
@Composable
fun ServerStatusView(isRunning: Boolean) {
    val dotColor = if (isRunning) BrandRunning else Color(0xFFBDBDBD)

    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Box(
            modifier = Modifier
                .size(56.dp)
                .clip(CircleShape)
                .background(dotColor.copy(alpha = 0.12f)),
            contentAlignment = Alignment.Center
        ) {
            Box(
                modifier = Modifier
                    .size(20.dp)
                    .clip(CircleShape)
                    .background(dotColor)
            )
        }

        Text(
            text = if (isRunning) "Running" else "Stopped",
            style = MaterialTheme.typography.titleSmall,
            fontWeight = FontWeight.SemiBold,
            color = if (isRunning) BrandSuccess else MutedText
        )
    }
}
